src/context.cpp

```diff
@@ -770,23 +651,23 @@ namespace librealsense
 
         void add_frame_metadata( frame * const f, json && dds_md )
         {
-            json const & md_header = dds_md["header"];
-            json const & md = dds_md["metadata"];
+            json const & md_header = dds_md[std::string( "header", 6 )];
+            json const & md = dds_md[std::string( "metadata", 8 )];
 
             // Always expected metadata
-            f->additional_data.timestamp = rsutils::json::get< rs2_time_t >( md_header, "timestamp" );
-            f->additional_data.timestamp_domain = rsutils::json::get< rs2_timestamp_domain >( md_header, "timestamp-domain" );
+            f->additional_data.timestamp = rsutils::json::get< rs2_time_t >( md_header, std::string( "timestamp", 9 ) );
+            f->additional_data.timestamp_domain
+                = rsutils::json::get< rs2_timestamp_domain >( md_header, std::string( "timestamp-domain", 16 ) );
 
             // Expected metadata for all depth images
-            if( rsutils::json::has( md_header, "depth-units" ) )
-                f->additional_data.depth_units = rsutils::json::get< float >( md_header, "depth-units" );
+            rsutils::json::get_ex( md_header, std::string( "depth-units", 11 ), &f->additional_data.depth_units );
```

Comment:
Will throw for color frames

Reply:
`get_ex` returns a bool on failure, and checks if it's there first. It can still throw, but not if it's not there.